request.c

```diff
@@ -854,23 +870,57 @@ void getForwardDestinationsOverTime(int *sock)
 	{
 		for(i = sendit; i < counters.overTime; i++)
 		{
+			double percentage;
+
 			validate_access("overTime", i, true, __LINE__, __FUNCTION__, __FILE__);
 			sprintf(server_message, "%i", overTime[i].timestamp);
 
-			int j;
+			int j, forwardedsum = 0;
+
+			// Compute forwardedsum used for later normalization
+			for(j = 0; j < overTime[i].forwardnum; j++)
+			{
+				forwardedsum += overTime[i].forwarddata[j];
+			}
 
+			// Loop over forward destinations to generate output to be sent to the client
 			for(j = 0; j < counters.forwarded; j++)
 			{
 				int k;
+
 				if(j < overTime[i].forwardnum)
+				{
+					// This forward destination does already exist at this timestamp
+					// -> use counter of requests sent to this destination
 					k = overTime[i].forwarddata[j];
+				}
 				else
+				{
+					// This forward destination does not yet exist at this timestamp
+					// -> use zero as number of requests sent to this destination
 					k = 0;
+				}
+
+				// Avoid floating point exceptions
+				if(forwardedsum > 0 && overTime[i].total > 0)
+				{
+					percentage = 1e2*k/forwardedsum*(overTime[i].total - (overTime[i].cached + overTime[i].blocked))/overTime[i].total;
```

Comment:
Can you clear up some of this math with some more parentheses? (particularly with relation to the division) And/or add a comment explaining the math.

Reply:
Parentheses are not needed here as `a/b*c/d == (a*c)/(b*d) == a*c/b/d`. However, I will add a description on how this works.

Reply:
It might be difficult to read the code without the parentheses because those rules might not be known, or aren't very intuitive.